lib/srv6.h

```diff
@@ -321,6 +321,7 @@ struct srv6_sid_ctx {
 	struct in_addr nh4;
 	struct in6_addr nh6;
 	vrf_id_t vrf_id;
+	ifindex_t ifindex;
```

Comment:
so nexthops have a ifindex already.  I don't understand enough about SRv6 to say if that would be ok or not to use?
For SRv6 what is the actual definition of a nexthop?  Does it possibly include a ifindex already?  What if they disagree?

Reply:
I updated the commit message to clarify this.

Reply:
would removing nh4 / nh6 vrf_id and ifindex, replacing with `struct nexthop` make anything clearer or easier to do?  I wonder why we have this specialized structure here, when we something generic like struct nexthop could work?

Reply:
Hi Donald,

Many thanks for your comments.

I’m already working on revisiting all the SRv6 data structures that have been added in the last years. I will open a refactoring PR for this.
I will definitely include your feedback in my refactoring PR.

I’m also doing some performance improvements. As part of the refactoring, I will also include these enhancements.